ext/drreg/drreg.c

```diff
@@ -863,19 +1391,248 @@ drreg_reserve_reg_internal(void *drcontext, instrlist_t *ilist, instr_t *where,
             __FUNCTION__, pt->live_idx, get_where_app_pc(where), get_register_name(reg),
             slot);
     }
-    pt->reg[GPR_IDX(reg)].native = false;
-    pt->reg[GPR_IDX(reg)].xchg = DR_REG_NULL;
-    pt->reg[GPR_IDX(reg)].slot = slot;
-    *reg_out = reg;
-    return DRREG_SUCCESS;
+    pt->reg[GPR_IDX(reg)].native = false;
+    pt->reg[GPR_IDX(reg)].xchg = DR_REG_NULL;
+    pt->reg[GPR_IDX(reg)].slot = slot;
+    *reg_out = reg;
+    return DRREG_SUCCESS;
+}
+
+#ifdef SIMD_SUPPORTED
+static void *
+get_simd_dead_state(const drreg_spill_class_t spill_class)
+{
+    if (spill_class == DRREG_SIMD_XMM_SPILL_CLASS) {
+        return SIMD_XMM_DEAD;
+    } else if (spill_class == DRREG_SIMD_YMM_SPILL_CLASS) {
+        return SIMD_YMM_DEAD;
+    } else if (spill_class == DRREG_SIMD_ZMM_SPILL_CLASS) {
+        return SIMD_ZMM_DEAD;
+    }
+    ASSERT(false, "cannot determine dead state");
+    return SIMD_UNKNOWN;
+}
+#endif
+
+#ifdef SIMD_SUPPORTED
+static drreg_spill_class_t
+get_spill_class(const reg_id_t reg)
+{
+
+    if (reg_is_gpr(reg)) {
+        return DRREG_GPR_SPILL_CLASS;
+    } else if (reg_is_strictly_xmm(reg)) {
+        return DRREG_SIMD_XMM_SPILL_CLASS;
+    } else if (reg_is_strictly_ymm(reg)) {
+        return DRREG_SIMD_YMM_SPILL_CLASS;
```

Comment:
Please add a TODO i#3844: comment that this is unsupported.

Reply:
Done, I added one single comment at the top to avoid clutter. I hope this is okay